salt/daemons/masterapi.py

```diff
@@ -548,6 +548,30 @@ def _mine_get(self, load, skip_verify=False):
         if not skip_verify:
             if any(key not in load for key in ('id', 'tgt', 'fun')):
                 return {}
+
+        if isinstance(load['fun'], six.string_types):
+            functions = list(set(load['fun'].split(',')))
+        else:
+            if not load.get('as_dict'):
+                log.warning('Passed function as object to mine.get but "as_dict" is {0}, '
```

Comment:
can you refactor those warnings to avoid duplicating the code?

I'm thinking to check if not load.get('as_dict') outside the else branch here and bundle its condition in the below check

Reply:
Make a placeholder for the error text you mean?